### PR TITLE
Provider Animekisa

### PR DESCRIPTION
**Pull request by Stormunblessed**
Animekisa.in (9anime.to scraper)
~~Pd: sometimes vidstream will throw bad https status, even though is the same link when inspecting~~

### CODE REVIEW

**Comment by cutterchop** [comment]
Hello, it looks like you used a rip off of the site and isnt the original site here is the like for the original

https://animekisa.tv/

I confirmed by checking there own discord server

**Comment by Stormunblessed** [comment]
> Hello, it looks like you used a rip off of the site and isnt the original site here is the like for the original
> 
> https://animekisa.tv/
> 
> I confirmed by checking there own discord server 

From what I testest animekisa.tv doesn't scrape from 9anime.to, that's why I added animekisa.in, 'cause it does scrape from 9anime.to

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimekisaProvider.kt [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            if (response.text.contains("#EXTM3U"))
+                M3u8Helper().m3u8Generation(
+                    M3u8Helper.M3u8Stream(
+                        response.url,
+                        headers = app.get(data).headers.toMap(),
+                    ), true
+                )
+                    .map { stream ->
+                        val qualityString = if ((stream.quality ?: 0) == 0) "" else "${stream.quality}p"
+                       listOf( callback( ExtractorLink(
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimekisaProvider.kt:145]: listOf?

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimekisaProvider.kt [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                            servername,
+                            "$servername $qualityString",
+                            stream.streamUrl,
+                            data,
+                            getQualityFromName(stream.quality.toString()),
+                            true
+                        )))
+                    }
+            }
+        //When the extracted vidstream link above won't work
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimekisaProvider.kt:155]: if you want both of these code blocks to run in parallel, you can use argpmap

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimekisaProvider.kt [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    override suspend fun loadLinks(
+        data: String,
+        isCasting: Boolean,
+        subtitleCallback: (SubtitleFile) -> Unit,
+        callback: (ExtractorLink) -> Unit
+    ): Boolean {
+        app.get(data).document.select("#servers-list ul.nav li a").apmap {
+            val server = it.attr("data-embed")
+            val servername = it.select("span").text()
+            val response = app.get(server, interceptor = WebViewResolver(
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimekisaProvider.kt:131]: If at all possible, then avoid webviewresolver

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimekisaProvider.kt [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+            Pair("$mainUrl/ajax/list/views?type=all", "All animes"),
+            Pair("$mainUrl/ajax/list/views?type=day", "Trending now"),
+            Pair("$mainUrl/ajax/list/views?type=week", "Trending by week"),
+            Pair("$mainUrl/ajax/list/views?type=month", "Trending by month"),
+
+            )
+
+        val items = ArrayList<HomePageList>()
+
+        for (i in urls) {
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimekisaProvider.kt:44]: you can use (url, name) in urls

File: app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimekisaProvider.kt [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        //When the extracted vidstream link above won't work
+        app.get(data).document.select("div#servers-list ul.nav li a:contains(VidStream)").apmap {
+            val server = it.attr("data-embed")
+            val getserver = app.get(server, referer = data).text
+            val key = getserver.substringAfter("window.skey = '").substringBefore("'")
+            val linkwkey = (server.replace("/e/","/info/")+"&skey=$key")
+            val normallink = app.get(linkwkey, referer = data).text
+            val extractedlink = normallink.substringAfter("},{\"file\":\"").substringBefore("\"}]")
+                .replace("\\/", "/")
+               if (extractedlink.isNotBlank()) M3u8Helper().m3u8Generation(
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/animeproviders/AnimekisaProvider.kt:164]: Does it always return m3u8?

**Review by LagradOst — APPROVED** [review]
Looks good 👍 Minor code changes and merge

**Comment by Stormunblessed** [comment]
Found out that WcoProvider and Animekisa.in are mirrors, so I made an extractor for Mcloud server and added it to Wco too

**Comment by LagradOst** [comment]
LGTM, ready to merge?

**Comment by Stormunblessed** [comment]
> LGTM, ready to merge?

Yeah

**Pull request merged**